### PR TITLE
PowerPC: Move lwarx/stwcxd. reservation into PowerPCState

### PR DESCRIPTION
**Pull request by CrystalGamma**

### CODE REVIEW

File: Source/Core/Core/PowerPC/PowerPC.cpp [hunk 1]
```diff
@@ -125,6 +125,17 @@ void DoState(PointerWrap& p)
   p.Do(ppcState.pagetable_base);
   p.Do(ppcState.pagetable_hashmask);
 
+  if (version >= 133)
```
Comment (leoetlino) [inline Source/Core/Core/PowerPC/PowerPC.cpp:128]: I'm not sure having backwards compatibility for savestates is worth the increased maintenance burden.

Comment (CrystalGamma) [reply Source/Core/Core/PowerPC/PowerPC.cpp:128]: I can take the commit back out if you want. Anyone else who might have an opinion on this?

File: Source/Core/Core/PowerPC/PowerPC.cpp [hunk 2]
```diff
@@ -125,6 +125,9 @@ void DoState(PointerWrap& p)
   p.Do(ppcState.pagetable_base);
   p.Do(ppcState.pagetable_hashmask);
 
+  p.Do(ppcState.reserve);
```
Comment (leoetlino) [inline Source/Core/Core/PowerPC/PowerPC.cpp:128]: The savestate version should be updated.

File: Source/Core/Core/PowerPC/PowerPC.cpp [hunk 3]
```diff
@@ -175,6 +178,7 @@ static void ResetRegisters()
   ppcState.pc = 0;
   ppcState.npc = 0;
   ppcState.Exceptions = 0;
+  ppcState.reserve = false;
```
Comment (leoetlino) [inline Source/Core/Core/PowerPC/PowerPC.cpp:181]: Maybe set reserve_address to 0 as well, just in case.

**Review by leoetlino — CHANGES_REQUESTED** [review]

File: Source/Core/Core/PowerPC/PowerPC.cpp (hunk 2 above)
Comment (CrystalGamma) [reply Source/Core/Core/PowerPC/PowerPC.cpp:128]: Right, forgot that while rolling back the compatibility code. Just a moment.

**Review by leoetlino — APPROVED** [review]

**Pull request merged**